test/secrets_test.rb

```diff
@@ -19,20 +19,29 @@ def process
       pemfile_path: 'vaultpem',
       ssl_verify: false,
       annotations: 'annotations',
-      output_path: Dir.pwd
+      serviceaccount_dir: Dir.pwd,
+      output_path: Dir.pwd,
+      api_url: 'https://foo.bar'
     )
   end
   let(:auth_reply) { {auth: {client_token: 'sometoken'}}.to_json }
+  let(:status_api_body) { {items: [{status: {hostIP: "10.10.10.10"}}]}.to_json }
 
   before do
     stub_request(:post, "https://foo.bar:8200/v1/auth/cert/login").
       to_return(body: auth_reply)
+    stub_request(:get, "https://foo.bar/api/v1/namespaces/default/pods").
+      to_return(body: status_api_body)
+    ENV["KUBERNETES_PORT_443_TCP_ADDR"] = 'foo.bar'
```

Comment:
use with_env / do not modify global env state without resetting it ... or set it at the top of the test file